framework/wazuh/core/agent.py

```diff
@@ -829,26 +789,31 @@ def _add_authd(self, name: str, ip: str, id: str = None, key: str = None, force:
         self.key = self.compute_key()
 
     @staticmethod
-    def delete_single_group(group_id: str) -> dict:
+    async def delete_single_group(group_name: str) -> dict:
         """Delete a group.
 
         Parameters
         ----------
-        group_id : str
-            Group ID.
+        group_name : str
+            Group name.
 
         Returns
         -------
         dict
             Confirmation message.
         """
-        # Delete group directory
-        group_path = path.join(common.SHARED_PATH, group_id)
-        if path.exists(group_path):
-            rmtree(group_path)
+        async with get_indexer_client() as indexer_client:
+            await indexer_client.agents.delete_group(group_name)
 
-        msg = "Group '{0}' deleted.".format(group_id)
+        # Delete group file
+        group_path = get_group_file_path(group_name)
+        if path.exists(group_path):
+            try:
+                remove(group_path)
+            except Exception as e:
+                raise WazuhError(1006, extra_message=str(e))
```

Comment:
Ideally this should be atomic, because if we delete the file and the indexer call fails we are in the same spot. I think network issues are more common than filesystem or permission issues, so I would prefer to leave it like this